itowns/Editing.js

```diff
@@ -127,17 +129,141 @@ class Editing {
+    this.checked = this.featureSelectedGeom.properties.status;
+    this.controllers.checked.updateDisplay();
+    this.viewer.comment = this.featureSelectedGeom.properties.comment;
+    this.controllers.comment.updateDisplay();
+  }
+
+  // alerts
+  async centerOnAlertFeature(onlyUnchecked = false, step = 0) {
+    const layerTest = this.viewer.view.getLayerById(this.alertLayerName);
+    const fc = await layerTest.source.loadData(undefined, layerTest);
+    this.nbValidated = fc.features[0].geometries.filter(
+      (elem) => elem.properties.status === true,
+    ).length;
+    this.nbTotal = fc.features[0].geometries.length;
+    this.nbChecked = `${this.nbValidated}/${this.nbTotal}`;
+    if (this.featureIndex === fc.features[0].geometries.length) this.featureIndex = 0;
+    if (this.featureIndex === -1) this.featureIndex = fc.features[0].geometries.length - 1;
+
+    if (onlyUnchecked) {
+      if (fc.features[0].geometries[this.featureIndex].properties.status === true) {
+        this.featureIndex += step;
+        if (this.featureIndex === fc.features[0].geometries.length) this.featureIndex = 0;
+        if (this.featureIndex === -1) this.featureIndex = fc.features[0].geometries.length - 1;
+      }
+    }
+
+    // Center on Feature
+    const coordcenter = fc.features[0].geometries[this.featureIndex].extent.clone()
+      .applyMatrix4(fc.matrixWorld).center();
+
+    itowns.CameraUtils.transformCameraToLookAtTarget(
+      this.viewer.view,
+      this.viewer.view.camera.camera3D,
+      {
+        coord: new itowns.Coordinates(this.viewer.crs, coordcenter.x, coordcenter.y),
+        heading: 0,
+      },
+    );
+    this.viewer.centerCamera(coordcenter.x, coordcenter.y);
+
+    this.highlightSelectedFeature(fc,
+      fc.features[0].geometries[this.featureIndex],
+      fc.features[0].type);
+  }
+
   keydown(e) {
     if (this.currentStatus === status.WAITING) return;
     console.log(e.key, ' down');
     if (this.currentStatus === status.RAS) {
       // L'utilisateur demande à déselectionner l'OPI
       if (this.validClicheSelected && (e.key === 'Escape')) {
         this.validClicheSelected = false;
-        this.cliche = 'unknown';
+        this.cliche = 'none';
         this.controllers.cliche.__li.style.backgroundColor = '';
         this.view.getLayerById('Opi').visible = false;
         this.view.notifyChange(this.view.getLayerById('Opi'), true);
+      } else if (this.alertLayerName && e.key === 'ArrowLeft') {
+        this.featureIndex -= 1;
+        this.centerOnAlertFeature();
+      } else if (this.alertLayerName && e.key === 'ArrowRight') {
+        this.featureIndex += 1;
+        this.centerOnAlertFeature();
+      } else if (this.alertLayerName && e.key === 'ArrowDown') {
+        this.featureIndex -= 1;
+        this.centerOnAlertFeature(true, -1);
+      } else if (this.alertLayerName && e.key === 'ArrowUp') {
+        this.featureIndex += 1;
+        this.centerOnAlertFeature(true, 1);
```

Comment:
Je trouve un peu étrange de faire un +/-1 sur l'index + d'avoir un paramètre step le le centerOnAlertFeature dans le cas où on navigue uniquement dans les alertes non sélectionnée. Est-ce qu'il ne serait pas plus simple de faire le +/-1 toujours dans centerOnAlertFeature?
Au passage, on pourrait renommer la fonction en centerOnNextAlertFeature?

Reply:
j'ai modifier la gestion de this.featureIndex. Et ai sorti les lignes de code liées presentes dans centerOnAlertFeature.